File: tlg/src/main/java/ch/bailu/tlg/HighscoreEntry.java
package ch.bailu.tlg;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.IOException;

public class HighscoreEntry {
    private final static int NAMEBUFFER_LIMIT=20;
    private final static String DEFAULT_NAME="-";

    public final String name;
    public final int score;

    
    
    public HighscoreEntry() {
        score = 0;
        name = DEFAULT_NAME;
    }

    
    public HighscoreEntry(String n, int s) {
        name = n;
        score = s;
        
    }
    
    
    public HighscoreEntry(BufferedInputStream reader) throws IOException {
        score = ByteInteger.read(reader);
        name = readName(reader);
    }

    
    private String readName(BufferedInputStream reader) throws IOException {
        byte[] buf;
        int len = ByteInteger.read(reader);

        if (len > NAMEBUFFER_LIMIT || len < 1) throw new IOException();
        buf= new byte[len];
        reader.read(buf);
        return new String(buf);
    }



    public void writeState(BufferedOutputStream output) throws IOException {
        System.err.print("write entry");
        System.err.print(name + "\n");

        
        byte[] buf=name.getBytes();
        ByteInteger.wrap(score).writeState(output);
        ByteInteger.wrap(buf.length).writeState(output);
        output.write(buf);
    }
}

File: tlg/src/main/kotlin/tlg/score/HighScoreEntry.kt
package tlg.score

import tlg.context.ByteInteger
import java.io.BufferedInputStream
import java.io.BufferedOutputStream
import java.io.IOException


class HighScoreEntry {
    val name: String
    val score: Int

    companion object {
        private const val NAME_BUFFER_LIMIT = 50
        private const val DEFAULT_NAME = "-"
    }

    constructor() {
        score = 0
        name = DEFAULT_NAME
    }

    constructor(n: String, s: Int) {
        name = n
        score = s
    }

    constructor(reader: BufferedInputStream) {
        score = ByteInteger.read(reader)
        name = readName(reader)
    }

    @Throws(IOException::class)
    private fun readName(reader: BufferedInputStream): String {
        val buf: ByteArray
        val len: Int = ByteInteger.read(reader)
        if (len > NAME_BUFFER_LIMIT || len < 1) throw IOException()
        buf = ByteArray(len)
        reader.read(buf)
        return String(buf)
    }

    @Throws(IOException::class)
    fun writeState(output: BufferedOutputStream) {
        println("write entry: $name - $score")
        ByteInteger.wrap(score).writeState(output)
        val nameBuffer = name.toByteArray()
        ByteInteger.wrap(nameBuffer.size).writeState(output)
        output.write(nameBuffer)
    }
}
